docs/features/shared-volumes-with-NFS.md

```diff
@@ -0,0 +1,138 @@
+# Supporting shared volumes with vSphere Docker Volumes
+
+## Requirements
+* Developer should able to use externally managed NFS share as “shared docker volume” using “vsphere” volume driver.
+* This allows them to manage NFS directories as docker volumes using Docker APIs, Compose etc 
+* Extend the vSphere Docker Volume plugin to support shared volumes
+* Volumes are accessible on multiple docker hosts running on  multiple ESX hosts in a Docker swarm
+* Allow user to specify what remote dirs are managed by the plugin
+* Support all current docker volume plugin API for shared volumes
+
+## User experience
+### Set up remote directories for docker volumes
+The design here assumes that the user is responsible to identify remote directories to export and has configured remote servers to export those. Specifically, the user needs to have,
+1. Exported remote directories to be used to host docker volumes.
+1. Configured sufficient permissions for the users on the Docker host to be able to mount and create folders and files on those directories.
+1. The docker host has the necessary packages of the desired version of NFS installed.
+1. The user can use the same remote directory configuration on multiple docker hosts on separate physical hosts and be able to view and use the same volumes on their containers (the applications must be able to synchronize access to shared volumes).
+
+### Update plugin configuration
+The plugin needs to be provided the details of the exported diretories and the servers from which to mount them. The details are entered into the plugin configuration file at /etc/docker-volume-vsphere.conf in the format as detailed later.
+
+### Restart plugin
+The plugin needs to be restarted to load the updated configuration.
+
+### Using shared volumes
+Docker volume commands must be used as shown below to use volumes on remote directories. Similar to how local VMDK based volumes are qualified with a datastore name, shared volumes must also be qualified with the label (see remote directory configuration below) that identifies the remote dir. to use to locate and refer a volume.
+1. docker volume create - 
+   1. _docker volume create --driver=vsphere --name=remote-vol -o type=file_ - uses the default entry in the list of remote dirs to locate the volume
+   1. _docker volume create --driver=vsphere --name=remote-vol@remote-dir-label_ - creates the volume in the location identified by the label in the remote dirs list
+1. docker volume list - No change
+1. docker volume inspect - the volume must be qualified by the remote dir label as below,
+   1. _docker volume inspect vol-name@remote-dir-label_
+1. docker run - the volume must be qualified by the remote dir label as below,
+   1. _docker run -it --volume-driver=vsphere -v remote-vol@remote-dir-label:/vol1 --name ub ubuntu_
+1. docker volume rm - the volume must be qualified by the remote dir label as below,
+   1. _dovker volume rm vol-name@remote-dir-label_
```

Comment:
This has to be handled by the NFS server that the dir. can't be removed if there are open files in the dir. CSI is handling this to ensure that the unmount is sent only if no containers are using the volume. Within the plugin prefer not to keep any meta-data to track this.